x/bucket/types/bucket.go

```diff
@@ -0,0 +1,78 @@
+package types
+
+import (
+	fmt "fmt"
+	io "io"
+	"strings"
+
+	"github.com/sonr-io/sonr/pkg/did"
+	"github.com/sonr-io/sonr/pkg/did/ssi"
+)
+
+func (b *Bucket) GetPath(address string, segments ...string) string {
+	path := fmt.Sprintf("/%s/%s", address, b.Uuid)
+	for _, segment := range segments {
+		path = fmt.Sprintf("%s/%s", path, segment)
+	}
+	return path
+}
+
+func (b *Bucket) GetDidService(address string) *did.Service {
+	segments := strings.Split(address, "snr")
+	service := fmt.Sprintf("did:snr:%s#%s", segments[1], b.Uuid)
+	return &did.Service{
+		ID:              ssi.MustParseURI(service),
+		Type:            "LinkedResource",
+		ServiceEndpoint: b.GetURI(address),
+	}
+}
+
+func (b *Bucket) GetURI(address string, items ...string) string {
+	params := NewParams()
+	bucketPath := b.GetPath(address)
+	path := fmt.Sprintf("%s/ipns/%s", params.IpfsGateway, bucketPath)
+	for _, item := range items {
+		path = fmt.Sprintf("%s/%s", path, item)
+	}
+	return path
+}
+
+type bucketItemImpl struct {
+	name    string
+	content []byte
+}
```

Comment:
does this mean buckets cannot reference other buckets? content is now an arbitrary `byte` array